app/src/main/java/com/github/orkest/View/profile/ProfileActivity.kt

```diff
@@ -22,28 +49,139 @@ class ProfileActivity() : ComponentActivity() {
         currentUser = intent.getStringExtra("username").toString()
         val viewModel = ProfileViewModel(currentUser)
         setContent {
-            OrkestTheme {
-                // A surface container using the 'background' color from the theme
-                Surface(
-                    modifier = Modifier.fillMaxSize(),
-                    color = MaterialTheme.colorScheme.background
-                ) {
-                    topProfile(viewModel = viewModel)
-                }
+            ProfileActivitySetting {
+                ProfileActivityScreen(this, viewModel = viewModel)
             }
         }
     }
 }
 
 @Composable
-fun topProfile(viewModel: ProfileViewModel) {
-    ProfileTopInterface(viewModel = viewModel)
+fun ProfileActivitySetting(content: @Composable () -> Unit) {
+    OrkestTheme {
+        // A surface container using the 'background' color from the theme
+        Surface(
+            modifier = Modifier.fillMaxSize(),
+            color = MaterialTheme.colorScheme.background
+        ) {
+            content()
+        }
+
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ProfileActivityScreen(activity: ComponentActivity, viewModel: ProfileViewModel) {
+    val scaffoldState = rememberScaffoldState()
+    val coroutineScope = rememberCoroutineScope()
+
+
+    androidx.compose.material.Scaffold(
+        // keep track of the state of the scaffold (whether it is opened or closed)
+        scaffoldState = scaffoldState,
+        // The content displayed inside the drawer when you click on the hamburger menu button
+        drawerContent = { CreateMenuDrawer() },
+        // main screen content
+        content = { padding ->
+            Modifier
+                .fillMaxHeight()
+                .padding(padding)
+            Column(
+                modifier = Modifier
+                    .fillMaxHeight()
+                    .padding(8.dp),
+            ) {
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                    horizontalArrangement = Arrangement.SpaceBetween
+                ) {
+                    topProfile(viewModel = viewModel, scaffoldState, coroutineScope)
+                }
+                mainBody()
+            }
+        },
+        drawerGesturesEnabled = true
+    )
+
+}
+
+@Composable
+fun topProfile(viewModel: ProfileViewModel, scaffoldState: ScaffoldState, coroutineScope: CoroutineScope) {
+    ProfileTopInterface(viewModel = viewModel, scaffoldState, coroutineScope)
+}
+
+@Composable
+fun mainBody() {
+    Divider(modifier = Modifier.padding(vertical = 10.dp))
+    favoriteSongs()
+    Divider(modifier = Modifier.padding(vertical = 10.dp))
+    favoriteArtists()
+}

```

Comment:
Maybe putting these values into variables would be better if future changes are needed